### PR TITLE
Adds ls troubleshooting tips

### PR DESCRIPTION
**Pull request by will-sargent-dbtlabs**
## What are you changing in this pull request and why?
The `dbt ls` command has been a personal goldmine find in helping to debug selectors and complex state:modified problems.

I wanted to get this documented for others to see examples of how it can be used for this purpose.  We do call out the command as an option in the table of commands, but there aren't any examples of how and why you might want to use it, and I believe it's arguably the best tool for troubleshooting this stuff, as it works in both the IDE and in jobs for logging.

## Checklist

- [X] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [X] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [X] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/reference/node-selection/syntax.md
Comment (mirnawong1) [inline website/docs/reference/node-selection/syntax.md]:
```suggestion
```

File: website/docs/reference/node-selection/syntax.md
Comment (mirnawong1) [inline website/docs/reference/node-selection/syntax.md:110]: hey @will-sargent-dbtlabs this is a great pr and what a wealth of knowledge 🙏 - thank you for sharing! I've tweaked it so it sits above the discourse community header and also added context as to what each command does - can you verify if that's technically right? i wasn't sure about the 2nd and 4th examples

File: website/docs/reference/node-selection/syntax.md
Comment (mirnawong1) [inline website/docs/reference/node-selection/syntax.md]: should it be this?
```suggestion
dbt ls --select "path/to/my/models" # Lists all models in a specific directory.
```

File: website/docs/reference/node-selection/syntax.md
Comment (mirnawong1) [inline website/docs/reference/node-selection/syntax.md:114]:
```suggestion
dbt ls --select "result:STATUS+ state:modified+ --defer --state ./dbt-artifact-path" # Lists nodes that match certain result statuses and are modified, which can help with troubleshooting.
```

File: website/docs/reference/node-selection/syntax.md
Comment (will-sargent-dbtlabs) [reply website/docs/reference/node-selection/syntax.md]: I copied that from another example on the page. But I agree that slashes makes more sense

Comment (will-sargent-dbtlabs) [reply website/docs/reference/node-selection/syntax.md]: I don't think dots actually work AFAIK
I got this from the "Examples of select flag" block up the page

Comment (mirnawong1) [reply website/docs/reference/node-selection/syntax.md]: awesome, thank yoU! i'll try to find the other one and update it too! 🙏

**Comment by will-sargent-dbtlabs** [comment]
Not sure on the statis one. I copied it from below again as well. We might need a @dbeatty10 or a core expert to weigh in..

File: website/docs/reference/node-selection/syntax.md
Comment (will-sargent-dbtlabs) [inline website/docs/reference/node-selection/syntax.md]: This is a core team question on if this is valid, because I lifted this from below as well..
Either way, if we removed it, no big deal as it was just a sample example of something complicated

Comment (will-sargent-dbtlabs) [reply website/docs/reference/node-selection/syntax.md]: Maybe that is a placeholder for SUCCESS or FAILED?

Comment (will-sargent-dbtlabs) [reply website/docs/reference/node-selection/syntax.md]: Yep:
result:error	✅	✅	✅	✅
result:success	✅	✅	✅	
result:skipped	✅		✅	✅
result:fail				✅
result:warn				✅
result:pass

Comment (will-sargent-dbtlabs) [reply website/docs/reference/node-selection/syntax.md]: So I think it's probably ok Maybe we need to tweak the table itself too..

Comment (will-sargent-dbtlabs) [reply website/docs/reference/node-selection/syntax.md]: Added a commit to do that

**Comment by mirnawong1** [comment]
this looks brilliant @will-sargent-dbtlabs , thank you so much! preview here: https://docs-getdbt-7nc6j3cyp-dbt-labs.vercel.app/reference/node-selection/syntax#troubleshoot-with-the-ls-command

File: website/docs/reference/node-selection/syntax.md
Comment (mirnawong1) [inline website/docs/reference/node-selection/syntax.md]:
```suggestion
dbt ls --select "result:<status>+ state:modified+ --defer --state ./<dbt-artifact-path>" # Lists nodes that match certain [result statuses](/reference/node-selection/syntax#the-result-status) and are modified.
```

**Review by mirnawong1 — APPROVED** [review]
🚢 and ready to merge!  happy to get @dbeatty10 's feedback before merging 🙏  thank you team!

**Review by will-sargent-dbtlabs — COMMENTED** [review]
LGTM

**Comment by mirnawong1** [comment]
ok going to merge this for you @will-sargent-dbtlabs ! we can come back to update it if needed. thank you so much!

**Pull request merged**
